sys/Makefile.include

```diff
@@ -88,6 +96,18 @@ ifneq (,$(filter scanf_float,$(USEMODULE)))
   endif
 endif
 
+ifeq (,$(filter printf_float,$(USEMODULE)))
```

Comment:
Both blocks could be factorized in a single one I think:
```suggestion
ifeq (,$(filter printf_float scanf_float,$(USEMODULE)))
```
Since the CFLAGS is the same for both printf and scanf.

Reply:
This code wasn't working because there was code in makefiles/libc/picolibc.mk that was overriding it. I've removed the code from sys/Makefile.include and fixed the code in makefiles/libc/picolibc.mk to check USEMODULE instead of using a separate configuration variable.